File: test/refmd/crypto/encoding_test.exs
defmodule RefMD.Crypto.EncodingTest do
  use ExUnit.Case, async: true

  alias RefMD.Crypto.Encoding

  describe "strict base64url encoding" do
    test "accepts only canonical unpadded base64url" do
      encoded = Encoding.encode_base64url("foo")
      assert encoded == "Zm9v"
      assert Encoding.decode_base64url!(encoded, 3) == "foo"
      assert_raise ArgumentError, fn -> Encoding.decode_base64url!("Zg==") end
      assert_raise ArgumentError, fn -> Encoding.decode_base64url!("Zm8/") end
      assert_raise ArgumentError, fn -> Encoding.decode_base64url!("A") end
    end
  end
end

File: assets/src/shared/lib/crypto/encoding.test.ts
import { describe, expect, it } from "vite-plus/test";
import { assertBase64Url, decodeBase64UrlStrict, encodeBase64Url } from "./encoding";

describe("strict base64url encoding", () => {
  it("accepts only canonical unpadded base64url", () => {
    const encoded = encodeBase64Url(new Uint8Array([102, 111, 111]));
    expect(encoded).toBe("Zm9v");
    expect(decodeBase64UrlStrict(encoded, 3)).toEqual(new Uint8Array([102, 111, 111]));
    expect(() => assertBase64Url("Zg==")).toThrow();
    expect(() => assertBase64Url("Zm8/")).toThrow();
    expect(() => assertBase64Url("A")).toThrow();
  });
});
